services/QuillLMS/spec/models/schools_users_spec.rb

```diff
@@ -33,6 +35,37 @@
     end
   end
 
+  describe '#reset_teacher_verification' do
+    context 'when user has teacher_info' do
+      let(:teacher_info) { create(:teacher_info, verification_status: TeacherInfo::APPROVED, verifier_id: 123) }
+      let(:user_with_teacher_info) { create(:user, teacher_info:) }
+      let(:schools_user) { create(:schools_users, user: user_with_teacher_info, school:) }
+
+      it 'resets teacher_info verification status and verifier_id after save' do
+        schools_user.save
+        teacher_info.reload
+        expect(teacher_info.verification_status).to be_nil
+        expect(teacher_info.verifier_id).to be_nil
+      end
+
+      it 'resets teacher_info verification status and verifier_id after destroy' do
+        schools_user.destroy
+        teacher_info.reload
+        expect(teacher_info.verification_status).to be_nil
+        expect(teacher_info.verifier_id).to be_nil
+      end
+    end
+
+    context 'when user does not have teacher_info' do
+      let(:user_without_teacher_info) { create(:user) }
+      let(:schools_user) { create(:schools_users, user: user_without_teacher_info, school:) }
```

Comment:
Minor potential efficiency: calling `create(:schools_users)` with no additional arguments will automatically generate a `create(:teacher)` and a `create(:school)` which should get you a user without teacher_info.  So I think you can simplify your `let` calls to just `let(:schools_user) { create(:schools_users) }`